contracts/base/Settings.sol

```diff
@@ -363,6 +370,75 @@ contract Settings is Pausable, SettingsInterface {
         );
     }
 
+    /**
+        @notice Add a new Node Component with its version.
+        @param componentName name of the component to be added.
+        @param minVersion minimum component version supported.
+     */
+    function createComponentVersion(bytes32 componentName, uint32 minVersion)
+        external
+        onlyPauser()
+        whenNotPaused()
+    {
+        require(minVersion > 0, "INVALID_COMPONENT_VERSION");
+        require(componentName != "", "COMPONENT_NAME_MUST_BE_PROVIDED");
+        require(componentVersions[componentName] == 0, "COMPONENT_ALREADY_EXISTS");
+        componentVersions[componentName] = minVersion;
+        emit ComponentVersionCreated(msg.sender, componentName, minVersion);
+    }
+
+    /**
+        @notice Remove a Node Component from the list.
+        @param componentName name of the component to be removed.
+     */
+    function removeComponentVersion(bytes32 componentName)
```

Comment:
Should a version be allowed to be removed? Seems like something we should keep a record of on-chain.